### PR TITLE
feat(checkbox): required state

### PR DESCRIPTION
**Pull request by JoaoFerreira-FrontEnd**
Issue number: resolves N/A

---------

## What is the new behavior?

- Adds the required state to ionic theme checkbox

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: core/src/components/checkbox/test/basic/checkbox.e2e.ts [hunk 1]
```diff
@@ -141,4 +141,21 @@ configs({ modes: ['ionic-md'], directions: ['ltr'] }).forEach(({ title, screensh
       await expect(checkboxes).toHaveScreenshot(screenshot(`checkbox-small`));
     });
   });
+
+  test.describe(title('checkbox: basic visual tests'), () => {
+    test('should have a Invalid visual applied correctly', async ({ page }) => {
```
Comment (sean-perkins) [inline core/src/components/checkbox/test/basic/checkbox.e2e.ts:146]:
```suggestion
    test('should have an invalid visual applied correctly', async ({ page }) => {
```

File: core/src/components/checkbox/test/basic/checkbox.e2e.ts [hunk 2]
```diff
@@ -141,4 +141,21 @@ configs({ modes: ['ionic-md'], directions: ['ltr'] }).forEach(({ title, screensh
...
+      await page.setContent(
+        `
+        <div id="checkboxes">
```
Comment (sean-perkins) [inline core/src/components/checkbox/test/basic/checkbox.e2e.ts:149]: We should add some padding to this container so the captured screenshot for the focus state includes the entire outline. 

For example:
```html
<div id="checkboxes" style="padding: 8px">
```

Would be good to double check the padding value to make sure it is large enough.

File: core/src/components/checkbox/checkbox.ionic.scss [hunk 3]
```diff
@@ -25,4 +27,20 @@
...
+// Ionic Design Checkbox Required
+// --------------------------------------------------
+:host(.ion-invalid) {
+  .checkbox-icon {
+    border-color: #f72c2c;
+  }
+}
+
+// Focused
+:host(.ion-focused.ion-invalid) .native-wrapper {
```
Comment (sean-perkins) [inline core/src/components/checkbox/checkbox.ionic.scss:41]: These styles build on the changes you have here: https://github.com/ionic-team/ionic-framework/pull/29286/files#diff-796ffc72b6171910c25d5aad085bc245095c38f104d55e253e06282702205066R77-R81.

If we wait for that PR to be reviewed first, this implementation can be simplified to:

```scss
:host(.ion-invalid) {
  --focus-ring-color: #ffafaf;
}
```

File: core/src/components/checkbox/test/basic/checkbox.e2e.ts (hunk 2 above)
Comment (JoaoFerreira-FrontEnd) [reply core/src/components/checkbox/test/basic/checkbox.e2e.ts]: I did test this, and it should be enough, thanks for the headsup

**Review by sean-perkins — APPROVED** [review]

**Pull request merged**